unittests/FEXLinuxTests/tests/syscalls/syscalls_efault.cpp

```diff
@@ -0,0 +1,163 @@
+#include <catch2/catch.hpp>
+
+#include <cstdint>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+#include <poll.h>
+#include <signal.h>
+
+TEST_CASE("poll") {
+  // poll can return EFAULT if first argument is pointed to invalid pointer.
+  struct pollfd *invalid_fds = reinterpret_cast<struct pollfd *>(mmap(nullptr, sysconf(_SC_PAGESIZE), PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0));
+  auto ret = ::syscall(SYS_poll, invalid_fds, 1, 0);
+  REQUIRE(ret == -1);
+  CHECK(errno == EFAULT);
+}
+
+TEST_CASE("ppoll") {
+  // ppoll can return EFAULT for arguements 1, 3, 4
+  struct pollfd *invalid_fds = reinterpret_cast<struct pollfd *>(mmap(nullptr, sysconf(_SC_PAGESIZE), PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0));
+  struct timespec *invalid_timespec = reinterpret_cast<struct timespec*>(mmap(nullptr, sysconf(_SC_PAGESIZE), PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0));
+  sigset_t *invalid_sigset = reinterpret_cast<sigset_t*>(mmap(nullptr, sysconf(_SC_PAGESIZE), PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0));
+
+  {
+    auto ret = ::syscall(SYS_ppoll, invalid_fds, 1, 0, nullptr, nullptr);
+    REQUIRE(ret == -1);
+    CHECK(errno == EFAULT);
+  }
```

Comment:
Since each of these blocks is an independent test, these are a good fit for SECTION:
```cpp
  SECTION("invalid FDs") {
    auto ret = ::syscall(SYS_ppoll, invalid_fds, 1, 0, nullptr, nullptr);
    REQUIRE(ret == -1);
    CHECK(errno == EFAULT);
  }
```

This will restart the test case for each SECTION, such that a failing REQUIRE won't cause the other sections to be skipped.

Reply:
Added sections